identity-tests/src/test/scala/com/gu/identity/integration/test/features/SocialNetworkTests.scala

```diff
@@ -82,6 +82,24 @@ class SocialNetworkTests extends IdentitySeleniumTestSuite {
       SocialNetworkSteps().deleteFacebookTestUser(facebookUser)
     }
 
+      scenarioWeb("should be asked to re-request e-mail permissions after denying them the first time") { implicit driver: WebDriver =>
+        val facebookUser = SocialNetworkSteps().createNewFacebookTestUser()
+        SocialNetworkSteps().goToFacebookAsUser(facebookUser)
+        BaseSteps().goToStartPage()
+        val registerPage = SignInSteps().clickSignInLink().clickRegisterNewUserLink()
+        val authDialog = registerPage.switchToNewSignIn().clickRegisterWithFacebookButton()
+        authDialog.clickEditInformationProvided().clickEmailCheckBox().clickConfirmButton()
+        SocialNetworkSteps().checkUserGotFacebookEmailError(registerPage)
+        SignInSteps().checkUserIsNotLoggedIn(facebookUser.fullName)
+        val signinPage = new SignInPage()
+        signinPage.clickFaceBookSignInButton(false)
+        val requestPermissionsDialog = new FaceBookAuthDialog()
+        requestPermissionsDialog.clickConfirmButton()
+        BaseSteps().goToStartPage()
+        SignInSteps().checkUserIsLoggedIn(facebookUser.fullName)
+        SocialNetworkSteps().deleteFacebookTestUser(facebookUser)
```

Comment:
Yeah, we have similar failure cases in all tests using the FB test user API.

Do you know if there is a way to do something similar to a `tearDown()` (something most other test frameworks have), essentially just running a method after each test has been executed to clean up state, regardless of whether the test failed or passed.